contents/df_end.md

```diff
@@ -141,36 +150,287 @@ filter(:name => ==("Alice"), grades_2020())
+
+```jl
+@sco salaries()
+```
+
+This data is about an imaginary situation where you want to figure out how many your colleagues earn, and haven't figured it out for Zed yet.
+Even though we don't want to encourage these practices, we suspect it is an interesting example.
+Say that we want to know who earns more than 2000.
+When using `filter` without taking the `missing` values into account, it will fail:
+
+```jl
+s = "filter(:salary => >(2_000), salaries())"
+sce(s, post=trim_last_n_lines(25))
+```
+
+`subset` will also fail, but guide us to an easy solution:
+
+```jl
+s = "subset(salaries(), :salary => ByRow(>(2_000)))"
+sce(s, post=trim_last_n_lines(25))
+```
+
+So, we just need to pass `skipmissing=true`:
+
+```jl
+s = "subset(salaries(), :salary => ByRow(>(2_000)); skipmissing=true)"
+sco(s; process=without_caption_label)
 ```
 
 ## Select {#sec:select}
 
-## Types and Missing Data, Categorical {#sec:missing_data}
+Whereas `filter` removes rows, `select` removes columns.
+However, this function is much more versatile than just removing columns, as we will descuss in this section.
+Lets create a dataset with multiple columns:
+
+```jl
+@sco responses()
+```
+
+Here, the data represents answers on five questions (`q1`, `q2`, ..., `q5`) in a questionnaire.
+Let's first "select" a few columns from this dataset.
+Again, we use symbols to specify columns:
+
+```jl
+s = "select(responses(), :id, :q1)"
+sco(s, process=without_caption_label)
+```
+
+But, we can also use strings if we want:
+
+```jl
+s = """select(responses(), "id", "q1", "q2")"""
+sco(s, process=without_caption_label)
+```
+
+To select everything _except_ one or more columns, use `Not` with a column name:
+
+```jl
+s = """select(responses(), Not(:q5))"""
+sco(s, process=without_caption_label)
+```
+
+Or, with multiple column names:
+
+```jl
+s = """select(responses(), Not([:q4, :q5]))"""
+sco(s, process=without_caption_label)
+```
+
+It's also fine to combine column names with `Not`:
+
+```jl
+s = """select(responses(), :q5, Not(:id))"""
+sco(s, process=without_caption_label)
+```
+
+Note how `q5` is now the first column.
+But, there is a more clever way to do this, and that is with `:`.
+The colon `:` can be thought of as "all the columns that we didn't include yet".
+For example:
+
+```jl
+s = """select(responses(), :q5, :)"""
+sco(s, process=without_caption_label)
+```
+
+Or, to put `q5` at the second position[^sudete]:
+
+[^sudete]: Thanks to Sudete on [Discourse](https://discourse.julialang.org/t/pull-dataframes-columns-to-the-front/60327/4) for this suggestion.
+
+```jl
+s = "select(responses(), 1, :q5, :)"
+sco(s, process=without_caption_label)
+```
+
+Even renaming is possible via select:
+
+```jl
+s = """select(responses(), 1 => "participant", :q1 => "age", :q2 => "nationality")"""
+sco(s, process=without_caption_label)
+```
+
+which, thanks to the `...` operator, we can also write as:
```

Comment:
Have we explained the splat operator? Or is its first appearance? Shall we need to cover in "Julia 101" chapter? Or in a footnote?

Reply:
https://github.com/JuliaDataScience/JuliaDataScience/issues/31